src/pages/Swap/index.tsx

```diff
@@ -796,7 +796,7 @@ export default function Swap({ className }: { className?: string }) {
                       </ButtonError>
                     </AutoColumn>
                   </AutoRow>
-                ) : permit.state === PermitState.PERMIT_NEEDED ? (
+                ) : isValid && permit.state === PermitState.PERMIT_NEEDED ? (
```

Comment:
nit: had to look up what `isValid` is referencing to understand this line

Reply:
it's not ideal but this file is a behemoth. i'm hoping to replace with widgets soon enough.